packages/contract/lib/contract/constructorMethods.js

```diff
@@ -119,6 +119,10 @@ module.exports = Contract => ({
   },
 
   async detectNetwork() {
+    // guard interfaceAdapter!
+    if (!Boolean(this.interfaceAdapter)) {
```

Comment:
This is just a longer way of saying `if (!this.interfaceAdapter)` that, IMO, doesn't really add any clarity to it.  I would suggest either the shorter way or the more explicit `if (this.interfaceAdapter == null)`.

Reply:
> Seems basically good, but I'm a bit confused about the reasoning for removing the messages... how would you get error messages if the tests _pass_ on an error? Since the failure case is when there's _no_ error, there won't be any pre-existing error message to pass on, right?
> 
Correct, no pre-existing error message to pass on, however there's still an indication that an expected failure didn't happen, like below where I simulated a failure in a test. I think this is acceptable, but will defer to you whether it's acceptable.

>   1) TruffleContract.deployed()
       throws if called before setting a provider:
     AssertionError [ERR_ASSERTION]: Missing expected rejection (Error).
      at async Context.<anonymous> (test/contract/constructorMethods.js:72:5)